testing/code/test_source.py

```diff
@@ -478,12 +478,13 @@ def deco_mark():
     def deco_fixture():
         assert False
 
-    src = inspect.getsource(deco_fixture)
+    # Since deco_fixture is now an instance of FixtureFunctionDef the getsource function will not work on it.
+    with pytest.raises(Exception):
+        inspect.getsource(deco_fixture)
+    src = inspect.getsource(deco_fixture.get_real_func())
     assert src == "    @pytest.fixture\n    def deco_fixture():\n        assert False\n"
-    # currently Source does not unwrap decorators, testing the
-    # existing behavior here for explicitness, but perhaps we should revisit/change this
```

Comment:
This test was introduced in https://github.com/pytest-dev/pytest/pull/6840/files and I updated it to work with the new code. Please let me know what you think.

Reply:
i wonder if it is a fair hack to set the `__wrapped__` attribute to a property that triggers a deprecationwarning